front/app/api/custom_fields/useCustomFields.ts

```diff
@@ -36,17 +61,15 @@ const useCustomFields = ({
         );
       });
 
-      const statementsForCustomField = statements.filter((statement) => {
-        const relationshipStatementIds =
-          customField.relationships.matrix_statements?.data.map(
-            (statement) => statement.id
-          );
-
-        return (
-          statement.data?.data.id &&
-          relationshipStatementIds?.includes(statement.data.data.id)
-        );
-      });
+      const statementsForCustomField =
+        statementsByCustomFieldId?.[customField.id] ?? [];
+      const convertedStatements: IMatrixStatementsType[] =
+        statementsForCustomField.map((statement) => ({
+          id: statement.id,
+          key: statement.attributes.key,
+          title_multiloc: statement.attributes.title_multiloc,
+          temp_id: statement.attributes.temp_id,
+        }));
```

Comment:
Refactored this to make it more understandable and to have proper type information